ansible/ocs.yaml

```diff
@@ -0,0 +1,292 @@
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Wait for storage nodes to be provisioned
+      shell: oc get nodes -l node-role.kubernetes.io/storage
+      retries: 100
+      delay: 30
+      register: storage_nodes_ready
+      until: (storage_nodes_ready.stdout | regex_findall('Ready    storage') | length) == 3
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Label storage nodes as local-storage capable
+      shell: |
+        for i in $(oc get nodes --no-headers -l node-role.kubernetes.io/storage -o name); do
+          oc label $i cluster.ocs.openshift.io/openshift-storage=''
+        done
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+    
+    - name: Deploy local-storage subscription
+      shell: |
+        oc new-project local-storage
+        oc annotate project local-storage --overwrite openshift.io/node-selector=''
+        oc apply -f {{ working_yamls_dir }}/ocs/local-storage-sub.yaml
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Wait for local-storage operator to be ready
+      shell: oc get pods -n local-storage
+      retries: 100
+      delay: 20
+      register: local_storage_operator_ready
+      until: (local_storage_operator_ready.stdout | regex_findall('local-storage-operator-.+-.+Running') | length) == 1
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Create local-storage PVs
+      shell: |
+        oc apply -f {{ working_yamls_dir }}/ocs/local-storage-volumes.yaml
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Wait for local-storage PVs to be ready
+      shell: oc get pv -n local-storage
+      retries: 100
+      delay: 20
+      register: local_storage_volumes_ready
+      until: (local_storage_volumes_ready.stdout | regex_findall('local-pv-.+Available') | length) == (local_storage_volumes_ready.stdout | regex_findall('local-pv-.+') | length) and (local_storage_volumes_ready.stdout | regex_findall('local-pv-.+Available') | length) != 0
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Deploy OCS subscription
+      shell: oc apply -f {{ working_yamls_dir }}/ocs/ocs-sub.yaml
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Wait for OCS, rook-ceph and noobaa operators to be ready
+      shell: oc get pods -n openshift-storage
+      retries: 100
+      delay: 20
+      register: ocs_operators_ready
+      until: (ocs_operators_ready.stdout | regex_findall('ocs-operator-.+-.+Running') | length) == 1 and (ocs_operators_ready.stdout | regex_findall('rook-ceph-operator-.+-.+Running') | length) == 1 and (ocs_operators_ready.stdout | regex_findall('noobaa-operator-.+-.+Running') | length) == 1
```

Comment:
Is there not some deployment you can wait on?

Reply:
That sounds like a better practice.  I'll see if I can do that for this and the final task.

Reply:
So I've tried switch all these various `regex_findall` checks to use `oc wait` instead, and to be honest it's not playing out that well.  There are a lot of transient errors that can pop-up during an `oc wait`, and adding `failed_when` or something similar to catch each and every one is making the tasks even uglier than they already were.  I found myself having to re-run the playbook again and again, often times encountering a new temporary error during a `wait` that stopped the playbook from proceeding.  Thoughts?